gpt_engineer/applications/cli/main.py

```diff
@@ -0,0 +1,216 @@
+    ),
+    lite_mode: bool = typer.Option(
+        False,
+        "--lite",
+        "-l",
+        help="Lite mode - run only the main prompt.",
+    ),
+    clarify_mode: bool = typer.Option(
+        False,
+        "--clarify",
+        "-c",
+        help="Lite mode - discuss specification with AI before implementation.",
+    ),
+    self_heal_mode: bool = typer.Option(
+        False,
+        "--self-heal",
+        "-sh",
+        help="Lite mode - discuss specification with AI before implementation.",
+    ),
+    azure_endpoint: str = typer.Option(
+        "",
+        "--azure",
+        "-a",
+        help="""Endpoint for your Azure OpenAI Service (https://xx.openai.azure.com).
+            In that case, the given model is the deployment name chosen in the Azure AI Studio.""",
+    ),
+    use_custom_preprompts: bool = typer.Option(
+        False,
+        "--use-custom-preprompts",
+        help="""Use your project's custom preprompts instead of the default ones.
+          Copies all original preprompts to the project's workspace if they don't exist there.""",
+    ),
+    verbose: bool = typer.Option(False, "--verbose", "-v"),
+):
+    logging.basicConfig(level=logging.DEBUG if verbose else logging.INFO)
+    #
+    if vector_improve_mode and not improve_mode:
+        print("Vector improve mode implies improve mode, setting improve_mode=True")
+        improve_mode = True
+
+    if improve_mode:
+        assert not (
+            clarify_mode or lite_mode
+        ), "Clarify and lite mode are not active for improve mode"
+
+    load_env_if_needed()
+
+    ai = AI(
+        model_name=model,
+        temperature=temperature,
+        azure_endpoint=azure_endpoint,
+    )
+
+    # project_path = os.path.abspath(
+    #     project_path
+    # )  # resolve the string to a valid path (eg "a/b/../c" to "a/c")
+    path = Path(project_path)  # .absolute()
+    print("Running gpt-engineer in", path.absolute(), "\n")
+    prompt = load_prompt(OnDiskRepository(path), improve_mode)
+    # configure generation function
+    if clarify_mode:
+        code_gen_fn = clarified_gen
+    elif lite_mode:
+        code_gen_fn = lite_gen
+    else:
+        code_gen_fn = gen_code
+    # configure execution function
+    if self_heal_mode:
+        execution_fn = self_heal
+    else:
+        execution_fn = execute_entrypoint
+
+    if --vector_improve_mode:
+        improve_fn = vector_improve
+    else:
+        improve_fn = improve
+
+    preprompts_path = get_preprompts_path(use_custom_preprompts, Path(project_path))
+    preprompts_holder = PrepromptsHolder(preprompts_path)
+    agent = CliAgent.with_default_config(
+        project_path,
+        ai=ai,
+        code_gen_fn=code_gen_fn,
+        execute_entrypoint_fn=execution_fn,
+        improve_fn=improve_fn,
+        preprompts_holder=preprompts_holder,
+    )
+    if improve_mode:
+        if --vector_improve_mode:
```

Comment:
shoud be "vector_improve_mode" not "--vector_improve_mode" unless im missing sth